engine/services/download_service.cc

```diff
@@ -179,195 +215,229 @@ cpp::result<bool, std::string> DownloadService::Download(
   curl_easy_cleanup(curl);
   return true;
 }
-void DownloadService::WorkerThread() {
+
+cpp::result<std::string, std::string> DownloadService::StopTask(
+    const std::string& task_id) {
+  // First try to cancel in queue
+  CTL_INF("Stopping task: " << task_id);
+  auto cancelled = task_queue_.cancelTask(task_id);
+  if (cancelled) {
+    EmitTaskStopped(task_id);
+    return task_id;
+  }
+  CTL_INF("Not found in pending task, try to find task " + task_id +
+          " in active tasks");
+  // Check if task is currently being processed
```

Comment:
nit: should add lock_guard in a block scope

Reply:
I will address this in following PR